### ISSUE TITLE
ProjectView / DeleteEvent not run when rebuilding projections

### ISSUE THREAD

**Issue by sturatcliffe** [JasperFx/marten#1302]
When using ProjectView to configure a DeleteEvent as documented here:

http://jasperfx.github.io/marten/documentation/events/projections/custom/

My projected document is deleted as expected. However, when rebuilding projections as documented here:

http://jasperfx.github.io/marten/documentation/events/projections/#sec3

My projected document appears again. It seems that ProjectView is not run when rebuilding projections. A sample console app to show this can be found here:

https://github.com/sturatcliffe/MartenProjectViewIssue

**Comment by oskardudycz** [comment]
@sturatcliffe just giving you heads up - I'm working on it, I hope that I'll come up with something this weak.

**Comment by ericgreenmix** [comment]
@oskardudycz Any update on your work on this? I hit this issue today.

**Comment by oskardudycz** [comment]
@ericgreenmix I was working on that, but it ended up with the refactoring. It's visible on draft PR https://github.com/JasperFx/marten/pull/1322. Although after discussion with @jeremydmiller we will probably go deeper with refactoring. See more here: https://github.com/JasperFx/marten/issues/1307 (you can put there also your comments about your thoughts/ideas).

If this is blocking you I can try to come up with some hotfix.

**Comment by ericgreenmix** [comment]
Cool. We are not majorly impacted by this at-the-moment. We only have one projection has the
```
opt.Events.InlineProjections.AggregateStreamsWith<Project>();
opt.Events.ProjectView<Project, Guid>().DeleteEvent<ProjectDeleted>();
```
format. I think we can actually convert it to a `ViewProjection`, not using `AggregateStreamsWith` and get around this issue.

I have been reading through #1307 and will post some thoughts when I get time tomorrow.

**Comment by oskardudycz** [comment]
@sturatcliffe @ericgreenmix please have a look on https://github.com/JasperFx/marten/pull/1483

It seems that the issue was fixed in one of the previous releases. I copied the repro from the linked project https://github.com/sturatcliffe/MartenProjectViewIssue.

Could you confirm that it works fine now?

**Comment by oskardudycz** [comment]
@sturatcliffe @ericgreenmix I managed to reproduce that. Fix will be available in `3.12.0`

**Comment by jeremydmiller** [comment]
Everybody, the mechanics are so different in V4 that I don't believe this issue matters. There is no `ProjectView()` in V4, and the delete events are working with the new `AggregateProjection`
